libs/nio/src/main/java/org/elasticsearch/nio/ChannelFactory.java

```diff
@@ -177,6 +177,7 @@ SocketChannel openNioChannel(InetSocketAddress remoteAddress) throws IOException
 
         SocketChannel acceptNioChannel(ServerChannelContext serverContext) throws IOException {
             ServerSocketChannel rawChannel = serverContext.getChannel().getRawChannel();
+            assert rawChannel.isBlocking() == false;
             SocketChannel socketChannel = accept(rawChannel);
             if (socketChannel == null) {
```

Comment:
Maybe also assert rawChannel.isBlocking() == false here? In case, this really is done by JVM behind our back somewhere?

Reply:
Hah yea let's do that, I like it :)